### PR TITLE
retsnoop: sidecar accepts verbs in requests

### PR DESCRIPTION
**Pull request by ThinkerYzu1**
The sidecar is going to provide various features.  To support that,
 every request have to be prefixed with a verb to designate what
 function it requests.

This change is on the top of PR3.

### CODE REVIEW

File: sidecar/src/main.rs [hunk 1]
```diff
@@ -77,6 +108,87 @@ fn print_function(name: &str, language: Option<gimli::DwLang>, demangle: bool) {
     }
 }
 
+fn query_address<T: gimli::Endianity>(addr: u64, ctx: &Context<gimli::EndianSlice<T>>, symbols: &SymbolMap<SymbolMapName>, config: &Config) {
```
Comment (anakryiko) [inline sidecar/src/main.rs:111]: please run `cargo fmt` on the project

**Review by anakryiko — COMMENTED** [review]
Looks great. Let's normalize the formatting and land PR #3 first, and this should be good to go (1 vs 11 is immaterial until we actually use that, but worth fixing).

File: sidecar/src/main.rs [hunk 2]
```diff
@@ -11,18 +11,25 @@ use std::fs::File;
...
+enum QueryType {
+    Addr(u64),
+    CompileUnit(String),
+}
+
+fn parse_query_line(string: &str) -> QueryType {
+    if string.len() > 10 && string.starts_with("symbolize ") {
+        QueryType::Addr(u64::from_str_radix(&string[10..], 16).expect("Failed to parse address"))
+    } else if string.len() > 11 && string.starts_with("query_syms ") {
+        QueryType::CompileUnit(string[1..].to_string())
```
[the remark this answers is not in the document]
Comment (ThinkerYzu1) [reply sidecar/src/main.rs:30]: Right! It is a typo. Should be 11.

File: sidecar/src/main.rs [hunk 3]
```diff
@@ -11,18 +11,25 @@ use std::fs::File;
...
-fn parse_uint_from_hex_string(string: &str) -> u64 {
-    if string.len() > 2 && string.starts_with("0x") {
-        u64::from_str_radix(&string[2..], 16).expect("Failed to parse address")
+enum QueryType {
+    Addr(u64),
+    CompileUnit(String),
+}
+
+fn parse_query_line(string: &str) -> QueryType {
+    if string.len() > 10 && string.starts_with("symbolize ") {
```
[the remark this answers is not in the document]
Comment (ThinkerYzu1) [reply sidecar/src/main.rs:27]: Right, it is not necssary.

**Review by anakryiko — COMMENTED** [review]
LGTM!

**Pull request merged**